packages/reporter/src/main.tsx

```diff
@@ -60,27 +74,63 @@ class Reporter extends Component<ReporterProps> {
   }
 
   render () {
-    const { appState } = this.props
+    const {
+      appState,
+      runMode,
+      runnablesStore,
+      scroller,
+      error,
+      events,
+      statsStore,
+      renderReporterHeader = (props: ReporterHeaderProps) => <Header {...props}/>,
+    } = this.props
 
     return (
-      <div className='reporter'>
-        <Header appState={appState} statsStore={this.props.statsStore} />
-        <Runnables
-          appState={appState}
-          error={this.props.error}
-          runnablesStore={this.props.runnablesStore}
-          scroller={this.props.scroller}
-          spec={this.props.spec}
-        />
+      <div className={cs('reporter', { multiSpecs: runMode === 'multi' })}>
+        {renderReporterHeader({ appState, statsStore })}
+        {this.props.runMode === 'single' ? (
+          <Runnables
+            appState={appState}
+            error={error}
+            runnablesStore={runnablesStore}
+            scroller={scroller}
+            spec={this.props.spec}
+          />
+        ) : this.props.allSpecs.map((spec) => (
+          <Runnables
+            key={spec.relative}
+            appState={appState}
+            error={error}
+            runnablesStore={runnablesStore}
+            scroller={scroller}
+            spec={spec}
+          />
+        ))}
+
         <ForcedGcWarning
           appState={appState}
-          events={this.props.events}/>
+          events={events}/>
       </div>
     )
   }
 
+  // this hook will only trigger if we switch spec file at runtime
+  // it never happens in normal e2e but can happen in component-testing mode
+  componentDidUpdate (newProps: BaseReporterProps) {
+    this.props.runnablesStore.setRunningSpec(this.props.spec.relative)
+
+    if (
+      this.props.resetStatsOnSpecChange &&
```

Comment:
It looks like we missed the changes from #14417

Reply:
Fixed ;)